### PR TITLE
Pass through task documentation

### PR DESCRIPTION
**Pull request by submitteddenied**
Added documentation for a tasks `next` callback.
Fix `api.tasks.allWorkingOn` docs.
Make examples consistent by removing parameters to `next` callback.

### CODE REVIEW

**Comment by submitteddenied** [comment]
I'd like to sweep back through here again and add more documentation/examples for the various queue inspection methods, but I think that's a PR for another day :smiley:

File: docs/core/tasks.md [hunk 1]
```diff
@@ -138,7 +147,7 @@ exports.sayHello = {
   frequency:     1000,
   run: function(api, params, next){
     api.log("hello")
-    next(null, true);
+    next();
```
Comment (evantahler) [inline docs/core/tasks.md]: There is a response signature here, and it is (error, data).  Data doesn't do much (just gets logged), but if you return an `error`, the job will be marked as failed

**Comment by evantahler** [comment]
Thanks for all your work!

Regrading the queue methods, you can link folks here: https://github.com/taskrabbit/node-resque#queue-managment

File: docs/core/tasks.md (hunk 1 above)
Comment (submitteddenied) [inline docs/core/tasks.md]: I do mention that higher up on Line 111, do you think an example of a job that fails (like a divide by with a param 0?) might be a better demonstration of failing a job like this?

Comment (evantahler) [inline docs/core/tasks.md]: Oh, I'm just noting that in documentation it's probably valuable to show all possible arguments, even if they are optional.

**Comment by submitteddenied** [comment]
How about this case, where the `sendWelcomeEmail` task passes the error through to fail the job if `sendEmail` fails. This also puts the example of failing a task close to where it is documented.

**Comment by evantahler** [comment]
Oh yeah! That should be updated to pass the `err` to next() as well.  Good call/catch

**Pull request merged**